src/NormalizedCollection/libs/RecordSetEventManager.js

```diff
@@ -72,7 +72,7 @@ RecordList.prototype = {
     util.log.debug('RecordList.add: key=%s, prevChild=%s', key, prevChild);
     var rec = this.obs.child(key);
     var fn = util.bind(this._valueUpdated, this, key);
-    this.loading[key] = {rec: rec, prev: prevChild, fn: fn};
+    this.loading[key] = {rec: rec, prev: prevChild, fn: fn, unwatch: function() { rec.unwatch('value', fn); }};
```

Comment:
Added an unwatch() function directly to the data we store to avoid complications with unwatching later.

The process looks like so:
  - store the rec/prevChild/and callback function in `this.loading` until the record is merged and returned
  - once it is returned, move the ref from `this.loading` into `this.recs` for monitoring
  - trigger a `child_added` event
  - trigger a `value` event

We also store and synchronize an ordered list of keys in `this.keys`. We use this to calculate the previous child key, since our `prevChild` from the Firebase event may not exist locally after we apply filtering.